### ISSUE TITLE
Migrate Meterpreter Error

### ISSUE THREAD

**Issue by umerov1999** [rapid7/metasploit-framework#9163]
meterpreter > migrate 7212
[*] Migrating from 4564 to 7212...
[-] Error running command migrate: Rex::TimeoutError Operation timed out

Ruby 2.4 and Ruby 2.3 Windows, doesn't work migrate Meterpreter(x32 -> x32 and x32 -> x64)

**Comment by busterb** [comment]
Darn it, https://github.com/rapid7/metasploit-payloads/pull/248/files fixed one bug, but created another.

**Comment by OJ** [comment]
Really!? Shit :(

**Comment by busterb** [comment]
Seems like it - you can switch to earlier payload build and verify. @bwatters-r7 would like to request migrate testing in the test automation if it's not there already :)

**Comment by umerov1999** [comment]
I tested on Ruby 2.4, now the process where the meterpreter migrates crashed

**Comment by OJ** [comment]
Someone else reported something similar in a version of Ruby running on Windows as well. But this might not be related to this issue if we've managed to cause migration to die. Wonder if it's Windows specific.

**Comment by busterb** [comment]
I found the issue and have partial patches for it. Will discuss with folks today and put up a PR.

**Comment by busterb** [comment]
See https://github.com/rapid7/metasploit-payloads/pull/250 above. Turning on debugging, I noticed that Windows Meterpreter starts spinning with high CPU when metasploit disconnects. This is due to it not properly handling the recv() failure and it retrying forever instead of bubbling up the error. 

The nested while loops combined with multiple break paths for both error handling and loop breakout was kind of insane, so I switched the error handling to just jump straight to the exception path instead.
